go/kbfs/libkbfs/tlf_handle_test.go

```diff
@@ -1269,3 +1277,168 @@ func TestParseTlfHandleImplicitTeams(t *testing.T) {
 	tid7, tlfID7 := newITeam("u1,u2#u3", "", tlf.Private)
 	check("u1,u2#u3", tid7, tlfID7, tlf.Private)
 }
+
+type offlineResolveCounterKBPKI struct {
+	KBPKI
+
+	lock                    sync.Mutex
+	bestEffortOfflineCounts map[string]int
+}
+
+func (d *offlineResolveCounterKBPKI) Resolve(
+	ctx context.Context, assertion string,
+	offline keybase1.OfflineAvailability) (
+	kbname.NormalizedUsername, keybase1.UserOrTeamID, error) {
+	if offline == keybase1.OfflineAvailability_BEST_EFFORT {
+		d.lock.Lock()
+		d.bestEffortOfflineCounts[assertion]++
+		d.lock.Unlock()
+	}
+	return d.KBPKI.Resolve(ctx, assertion, offline)
+}
+
+type testOfflineStatusPathsGetter struct {
+	bestEffortPaths map[string]bool
+}
+
+func (t *testOfflineStatusPathsGetter) OfflineAvailabilityForPath(
+	tlfPath string) keybase1.OfflineAvailability {
+	if t.bestEffortPaths[tlfPath] {
+		return keybase1.OfflineAvailability_BEST_EFFORT
+	}
+	return keybase1.OfflineAvailability_NONE
+}
+
+func (t *testOfflineStatusPathsGetter) OfflineAvailabilityForID(
+	tlfID tlf.ID) keybase1.OfflineAvailability {
+	panic("Not supported")
+}
+
+func TestParseTlfHandleOfflineAvailability(t *testing.T) {
+	ctx := context.Background()
+
+	localUsers := MakeLocalUsers([]kbname.NormalizedUsername{"u1", "u2", "u3"})
+	localUsers[0].Asserts = []string{"u1@twitter"}
+	currentUID := localUsers[0].UID
+	localTeams := MakeLocalTeams(
+		[]kbname.NormalizedUsername{"u1u2u3", "u3u2u1"})
+	daemon := NewKeybaseDaemonMemory(
+		currentUID, localUsers, localTeams, kbfscodec.NewMsgpack())
+
+	kbpki := &offlineResolveCounterKBPKI{
+		KBPKI: &daemonKBPKI{
+			daemon: daemon,
+		},
+		bestEffortOfflineCounts: make(map[string]int),
+	}
+
+	osg := &testOfflineStatusPathsGetter{make(map[string]bool)}
+	osg.bestEffortPaths["/keybase/private/u2"] = true
+
+	t.Log("Check unsynced private TLF")
+	_, err := ParseTlfHandle(ctx, kbpki, nil, osg, "u1", tlf.Private)
+	require.NoError(t, err)
+	require.Equal(t, kbpki.bestEffortOfflineCounts["u1"], 0)
```

Comment:
I didn't quite dig into the call stack here, but is it possible that these counts would be updated racily, such that they wouldn't be ready yet at this point?

Reply:
No that shouldn't happen.  `ParseTlfHandle` waits for all goroutines doing the resolves to complete.  But let me know if you had a specific scenario in mind that I'm missing.